### PR TITLE
WIP: Support running kops without an SSH public key

### PR DESCRIPTION
**Pull request by justinsb**
Issue #2659

### CODE REVIEW

**Comment by justinsb** [comment]
WIP because we need a way to know if the SSH public key omission was a mistake.  I suspect that 9/10 it will be a mistake.  @chrislovecnm what is the use-case for not being able to have a public key?

**Comment by justinsb** [comment]
Marking as kops-server because I expect we'll be able to address this when we address how we manage keys in kops-server

File: upup/pkg/fi/cloudup/apply_cluster.go [hunk 1]
```diff
@@ -378,13 +378,14 @@ func (c *ApplyClusterCmd) Run() error {
 				//"dnsZone": &awstasks.DNSZone{},
 			})
 
+			// TODO: We need to figure out how to safely skip this check when users really don't want an SSH key
```
Comment (chrislovecnm) [inline upup/pkg/fi/cloudup/apply_cluster.go]: We probably need an override value in the API.  `doNotConfigureSSHKey` or something like that.

**Comment by chrislovecnm** [comment]
Travis is failing:

```
--- FAIL: TestComplex (0.02s)
	integration_test.go:159: error running update cluster "complex.example.com": SSH public key must be specified when running with AWS (create with `kops create secret --name complex.example.com sshpublickey admin -i ~/.ssh/id_rsa.pub`)
I0718 01:45:25.034606   17167 zones.go:73] ListHostedZonesPages {
```

**Comment by chrislovecnm** [comment]
Any update on this @justinsb

**Pull request closed without merging**